src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -120,7 +120,14 @@ public void Add(MessageEndpointMapping mapping)
         /// </summary>
         protected override void BaseAdd(ConfigurationElement element)
         {
-            BaseAdd(element, true);
+            try
+            {
+                BaseAdd(element, true);
+            }
+            catch (ConfigurationErrorsException e)
+            {
+                throw new Exception($"An ambigious message endpoint mapping has been defined at line: {e.Line}. Check the 'MessageEndpointMappings' section in {e.Filename} for conflicting mappings.", e);
```

Comment:
> Although I like the idea we would couple this to another configuration property which might fail as well. Is this worth it?

so you're suggesting to not provide further details (line, filename, ...) and point at the inner exception instead?

> When something in configuration goes wrong I would actually expect a ConfigurationErrorsException. So instead of throwing a generic exception we could also throw a ConfigurationErrorsException wrapping the other one.

Not sure if I agree. I think it's very common in NSB that we wrap exceptions in a generic `Exception` for startup issues. Not sure whether you explicitly want to catch ConfigurationErrorsException and why the inner exception isn't good enough (since we also would have to provide the correct properties?)